doc/user-manual/language/built-ins.lagda.rst

```diff
@@ -273,21 +275,29 @@ numbers_>`_, `Bool <Booleans_>`_, `String <Strings_>`_ and `Int
     primATan2         : Float → Float → Float
     primShowFloat     : Float → String
 
-These are implemented by the corresponding Haskell functions with a few
-exceptions:
+The ``primFloatEquality`` primitive is intended to be used for decidable
+propositional equality. To enable proof carrying comparisons while preserving
+consisteny, the following laws apply:
 
 - ``primFloatEquality NaN NaN`` returns ``true``.
-- ``primFloatEquality NaN (primFloatNegate NaN)`` returns ``false``.
+- ``primFloatEquality NaN (primFloatNegate NaN)`` returns ``true``.
 - ``primFloatEquality 0.0 -0.0`` returns ``false``.
-- ``primFloatLess NaN NaN`` returns ``false``.
-- ``primFloatLess (primFloatNegate NaN) (primFloatNegate NaN)`` returns ``false``.
-- ``primFloatLess NaN (primFloatNegate NaN)`` returns ``false``.
-- ``primFloatLess (primFloatNegate NaN) NaN`` returns ``true``.
-- ``primFloatLess`` sorts ``NaN`` below everything but negative infinity.
-- ``primFloatLess -0.0 0.0`` returns ``true``.
-
-This is to allow decidable equality and proof carrying comparisons on floating
-point numbers.
+
+
+For numerical comparisons, use the ``primFloatNumericalEquality``
+and ``primFloatNumericalLess`` primitives. These are implemented by the
+corresponding Haskell functions with the following exceptions:
+
+- ``primFloatNumericalEquality 0.0 -0.0`` returns ``true``.
```

Comment:
I guess we should also say whether ``primFloatNumericalEquality NaN NaN`` returns true or not. I dont really care.